docs/supported-tools.rst

```diff
@@ -0,0 +1,260 @@
+we need to change our code as well to keep our features working.
+This is done with backwards compatibility in mind,
+but sometimes is hard to keep compatibility with old and new versions at the same time.
+In order to continue moving forward on future development and features,
+we need to drop support for some versions.
+
+.. note::
+
+   Your existent documentation will always be kept online and working,
+   but **when support for a tool or dependency ends, new builds may fail**.
+
+.. contents:: Contents
+   :local:
+   :depth: 3
+
+End of support policy
+---------------------
+
+Our policy defines how long a given tool or dependency is considered supported.
+Read the Docs will contact all users when an end of support date is close,
+after that date your builds may start failing and you will need to upgrade in order to receive support.
+For :doc:`/commercial/index` we provide an extended support of six months after the official end of support date.
+
+This is how we choose an end of support date:
+
+- For tools that define an :abbr:`EOL (End Of Life)` date, we try to follow that date for our policy.
+- For tools that release their versions incrementally without an EOL date,
+  we choose a date based on:
+
+  - The release versions with breaking changes (mayor versions).
+  - The date since it was last updated.
+  - And its usage on our platform.
+
+.. note::
+
+   Some recent versions may be supported, but aren't listed on these tables
+   (specially minor updates from supported versions).
+   Contact us if you have doubts.
+
+Default dependencies
+--------------------
+
+We install some dependencies in order to build your project without specifying their dependencies.
+In the past we used to set these dependencies to a specific version and update them after some time,
+but by doing so would break some builds and new projects wouldn't be able to use a new version by default.
+For this reason we are now installing their latest version (or latest supported version) by default.
+
+.. note::
+
+   In order to keep your builds reproducible,
+   it's highly recommended declaring its dependencies and versions explicitly.
+
+   .. TODO: link to this guide once it's written https://github.com/readthedocs/readthedocs.org/issues/7852.
+
+External dependencies (Python)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These are the dependencies that are installed by default when using a Python environment:
+
+Sphinx:
+  Projects created before Oct 20, 2020 use ``1.8.x``.
+  New projects use the latest version.
+
+Mkdocs:
+  Projects created before April 3, 2019 (April 23, 2019 for :doc:`/commercial/index`) use ``0.17.3``.
+  New projects use the latest version.
+
+sphinx-rtd-theme:
+  Projects created before October 20, 2020 (January 21, 2021 for :doc:`/commercial/index`) use ``0.4.3``.
+  New projects use the latest version.
+
+pip:
+  Latest version by default.
+
+setuptools:
+  Latest version by default.
```

Comment:
I just found out that this is required for Python 3.9 (#7885).

Using the version `41.0.1` (current) it fails when installing `mock==1.0.1` dependency.

We have a feature flag, `INSTALL_LATEST_SETUPOOLS` that works as a workaround for now, tho.

Reply:
This is already by default now.